crates/core/src/config.rs

```diff
@@ -93,8 +122,39 @@ pub struct LogConfig {
     pub directives: Vec<String>,
 }
 
-impl ConfigFile {
-    pub fn read(path: &ConfigToml) -> anyhow::Result<Option<Self>> {
-        parse_config(path.as_ref())
+/// Patch the value of a key in a TOML document,
+/// preserving the formatting and comments of the original value.
+fn patch_value(item: Option<&Item>, value: Option<&str>) -> Option<Item> {
+    match (value, item) {
+        (Some(value), Some(Item::Value(Value::String(v)))) => {
+            let mut new = Value::String(Formatted::new(value.to_string()));
+            let decor = new.decor_mut();
+            *decor = v.decor().clone();
+            Some(new.into())
+        }
+        (Some(val), _) => Some(val.into()),
+        (None, _) => None,
+    }
+}
+
+/// Set the value of a key in a `TOML` document, removing the key if the value is `None`.
+///
+/// **NOTE**: This function will preserve the formatting and comments of the original value.
+pub fn set_opt_value(doc: &mut DocumentMut, key: &str, value: Option<&str>) {
+    if let Some(value) = patch_value(doc.get(key), value) {
+        doc[key] = value;
+    } else {
+        doc.remove(key);
+    }
+}
+
+/// Set the value of a key in a `TOML` table, removing the key if the value is `None`.
+///
+/// **NOTE**: This function will preserve the formatting and comments of the original value.
+pub fn set_table_opt_value(table: &mut Table, key: &str, value: Option<&str>) {
```

Comment:
nit: why are these functions in this file? It looks like nothing else in this file has anything to do with tomls, so it's a bit surprising that they're added here.

Reply:
This file does contain some code that *reads* TOML files. See e.g. `MetadataFile::read` above. That said, the only place we ever *write* TOML files AFAICT is from the CLI, so I think these functions for writing TOML should be moved into the CLI code.

Reply:
Good point

Reply:
I consider that keeping all the stuff that touches `toml` in one place makes life easier.